src/libpgmoneta/json.c

```diff
@@ -844,6 +861,36 @@ fill_value(char* str, char* key, uint64_t* index, struct json* o)
    return 1;
 }
 
+static int
+handle_escape_char(char* str, uint64_t* index, uint64_t len, char* key)
+{
+   uint64_t idx = *index;
+   idx++;
+   if (idx == len) // security check
+   {
+      return 1;
+   }
+   switch (str[idx])
+   {
+      case '\'': // single quote
+         key = pgmoneta_append_char(key, '\'');
+         break;
+      case '\"': // double quote
+         key = pgmoneta_append_char(key, '\"');
+         break;
+      case '\n': // line feed
```

Comment:
This doesn't look right, have you tested it?

Reply:
Yeah I know, still working on it. I’ll ping you once done.

Reply:
Yeah, It will not capture any escape character at the starting of the string. (since key is NULL).

For that I am thinking of taking the corresponding escape character (after parsing) outside of `handle_escape_char` and then append it.

Reply:
No you missed the point. The string comes in as something like "\n", now you've seen the '\', so in the switch cases you should be checking for 'n' or 't', not '\n' or '\t'.

>  it will not capture any escape character at the starting of the string. (since key is NULL).

If you are talking about `pgmoneta_append_char()`, the function works even with NULL string.

Reply:
> No you missed the point. The string comes in as something like "\n", now you've seen the `'\'`, so in the switch cases you should be checking for 'n' or 't', not '\n' or '\t'.
> 

Yeah, this is wrong too, I might have missed it because while testing it, it worked totally fine.

It should have given me -
```
ERROR main.c:830 Management: Bad payload (1)
```
but actually it is not even entering the `handle_escape_char` function. Maybe while parsing `\n` is treated as single character.

Anyways will make the change. Thanks for pointing it!

> > it will not capture any escape character at the starting of the string. (since key is NULL).
> 
> If you are talking about `pgmoneta_append_char()`, the function works even with NULL string.

True, but in a case where we are sending an escape character at the starting of the string and passing pointer of `key/value` (which don't point to anything --> NULL) to the function `handle_escape_char` will not affect the value of the pointer and so it will still remain `NULL`.

Because in the caller the pointer was NULL and in the callee (where we are using malloc (realloc will call malloc if NULL is provided)) its reference changed.

Reply:
> but actually it is not even entering the handle_escape_char function. Maybe while parsing \n is treated as single character.

Right, so if you want to build a test cases in C you should use something like "\\n" to escape the backlash, you need to check other characters as well.

> Because in the caller the pointer was NULL and in the callee (where we are using malloc (realloc will call malloc if NULL is provided)) its reference changed.

That's a good point, the current implemention looks correct.

I think we still need `\\` and `\r`. Be careful with cases like `"\\""`, which would be illegal since the backslash itself is escaped.

Reply:
> 
> Right, so if you want to build a test cases in C you should use something like `"\\n"` to escape the backlash, you need to check other characters as well.
> 

What I am actually thinking is, while parsing if the character is treated as `\n` it won't simply go inside the handle_escape_character but will be stored as desired, while if we encounter `\n` as two separate characters `\` and `n` (for example in case where we are taking input from stdin or reading from a file) in that case it will get inside `handle_escape_character` and we will simply have to store `\n` character in place of `\n` (both characters treated individually).

> 
> That's a good point, the current implemention looks correct.
> 
> I think we still need `\\` and `\r`. Be careful with cases like `"\\""`, which would be illegal since the backslash itself is escaped.

In the my latest draft I have added checks for illegal use case.

Reply:
> What I am actually thinking is, while parsing if the character is treated as \n it won't simply go inside the handle_escape_character but will be stored as desired, while if we encounter \n as two separate characters \ and n (for example in case where we are taking input from stdin or reading from a file) in that case it will get inside handle_escape_character and we will simply have to store \n character in place of \n (both characters treated individually).

Yes, but you still need to test it and make sure it works.

> In the my latest draft I have added checks for illegal use case.

Sounds good, just push when it's ready.

Reply:
@Jubilee101 PTAL